src/components/NcAppNavigation/NcAppNavigation.vue

```diff
@@ -64,7 +64,10 @@ emit('toggle-navigation', {
 			class="app-navigation__content"
 			:inert="!open || undefined"
 			@keydown.esc="handleEsc">
-			<slot />
+			<div v-if="$scopedSlots.default" class="app-navigation__body">
+				<slot />
+			</div>
+
```

Comment:
With the default slot, it is always more complicated. If not `<template #default>` is used, then the slot is passed. Even it is actually empty with empty strings from the template.

![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/25978914/f8a40849-9af7-417b-9766-f7389902a76e)

Reply:
`$slots` will work here in Vue 2 as well as double-render with `isSlotPopupated()` in Vue 3

Reply:
And because of flex, it is stretched by default:

![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/25978914/94be44b2-5246-426b-90b8-75e977e8e08f)

Reply:
> With the default slot, it is always more complicated. If short syntax (not `<template #default>`) is used, then the slot is passed. Even it is actually empty with empty strings from the template.

In this case it seems to not be an empty string but rather a modal https://github.com/nextcloud/server/blob/c18ffe0cad13dc5903cbb08ffbdb787f01db5e35/apps/files/src/views/Navigation.vue#L71-L73 that is causing this 🤔

Not sure how we could resolve this on nc/vue side, any ideas?

> And because of flex, it is stretched by default:

How can I repro this?

Reply:
> In this case it seems to not be an empty string but rather a modal https://github.com/nextcloud/server/blob/c18ffe0cad13dc5903cbb08ffbdb787f01db5e35/apps/files/src/views/Navigation.vue#L71-L73 that is causing this 🤔

Yes, but the same issue would be in other cases. Anything in components's content, even if it actually is not rendered, cause this issue. With the default slot it's always more annoying...

> Not sure how we could resolve this on nc/vue side, any ideas?

Let me have a look on the issue...

> How can I repro this?

I though that was on this branch + your branch on server for the same issue. Be let me double check.

Reply:
> > And because of flex, it is stretched by default:
> 
> How can I repro this?

Ok, it seems to work fine now on your branch. But I'm not sure if it is not a regression for other usages.